srtcore/core.cpp

```diff
@@ -4922,10 +4923,12 @@ EConnectStatus srt::CUDT::postConnect(const CPacket* pResponse, bool rendezvous,
     //int token = -1;
 #if ENABLE_BONDING
     {
-        ScopedLock cl (uglobal().m_GlobControlLock);
+        SharedLock cl (uglobal().m_GlobControlLock);
         CUDTGroup* g = m_parent->m_GroupOf;
         if (g)
         {
+            // TODO: Lock this group `g`??
```

Comment:
Yes, that's likely a good idea. And also the shared lock suffices for that, while `acquireSocketsGroup` could be used here (that is, the constructor overload of GroupKeeper that uses this call).